### PR TITLE
Task/Hide dependencies & refactor

### PR DESCRIPTION
**Pull request by solid-danylokhvan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,31 +51,17 @@ import 'package:language_tool_textfield/language_tool_textfield.dart';
...
-final _debouncedLangService = DebounceLangToolService(
-  LangToolService(_languageTool),
-  const Duration(milliseconds: 500),
-);
-
 // Create a text controller for the Widget
-final _controller = ColoredTextEditingController(
-    languageCheckService: _debouncedLangService
-);
+final _controller = ColoredTextEditingController();
```
Comment (illia-romanenko) [inline README.md:55]: LanguageToolController

Comment (illia-romanenko) [reply README.md:55]: Also please provide somewhere note about debounce in readme. It doesn't have to be the same section.

**Comment by solid-danylokhvan** [comment]
Снимок экрана 2023-07-14 в 18 37 24

**Comment by solid-danylokhvan** [comment]
Снимок экрана 2023-07-14 в 18 57 31

File: example/lib/main.dart [hunk 2]
```diff
@@ -20,24 +20,8 @@ class App extends StatefulWidget {
...
-  static final DebounceLangToolService _debouncedLangService =
-      DebounceLangToolService(
-    LangToolService(_languageTool),
-    const Duration(milliseconds: 500),
-  );
-
-  /// Initialize ColoredTextEditingController
-  final ColoredTextEditingController _controller =
-      ColoredTextEditingController(languageCheckService: _debouncedLangService);
+  /// Initialize LanguageToolTextEditingController
```
Comment (illia-romanenko) [inline example/lib/main.dart:23]:
```suggestion
  /// Initialize LanguageToolController
```

File: example/lib/main.dart [hunk 3]
```diff
@@ -54,10 +38,12 @@ class _AppState extends State<App> {
...
             LanguageToolTextField(
-              style: const TextStyle(),
-              decoration: const InputDecoration(),
-              coloredController: _controller,
-              mistakePopup: MistakePopup(popupRenderer: PopupOverlayRenderer()),
+              controller: _controller,
+
+              // A language code like en-US, de-DE, fr, or auto to guess
+              // the language automatically.
+              // language = 'auto' by default.
```
Comment (illia-romanenko) [inline example/lib/main.dart:45]:
```suggestion
```

It's in the docs anyway

**Review by illia-romanenko — APPROVED** [review]
Please fix suggestions and I think we can merge.

**Pull request merged**